src/Symfony/Component/Cache/Adapter/RecordingAdapter.php

```diff
@@ -0,0 +1,194 @@
+        $time = microtime(true) - $start;
+
+        $object = (object) compact('name', 'arguments', 'start', 'time', 'result');
+
+        return $object;
+    }
+
+    public function getItem($key)
+    {
+        $call = $this->timeCall(__FUNCTION__, array($key));
+        $result = $call->result;
+        $call->isHit = $result->isHit();
+
+        // Display the result in a good way depending on the data type
+        if ($call->isHit) {
+            $call->result = $this->getValueRepresentation($result->get());
+        } else {
+            $call->result = null;
+        }
+
+        $this->addCall($call);
+
+        return $result;
+    }
+
+    public function hasItem($key)
+    {
+        $call = $this->timeCall(__FUNCTION__, array($key));
+        $this->addCall($call);
+
+        return $call->result;
+    }
+
+    public function deleteItem($key)
+    {
+        $call = $this->timeCall(__FUNCTION__, array($key));
+        $this->addCall($call);
+
+        return $call->result;
+    }
+
+    public function save(CacheItemInterface $item)
+    {
+        $key = $item->getKey();
+        $value = $this->getValueRepresentation($item->get());
+
+        $call = $this->timeCall(__FUNCTION__, array($item));
+        $call->arguments = array('<CacheItem>', $key, $value);
+        $this->addCall($call);
+
+        return $call->result;
+    }
+
+    public function saveDeferred(CacheItemInterface $item)
+    {
+        $key = $item->getKey();
+        $value = $this->getValueRepresentation($item->get());
+
+        $call = $this->timeCall(__FUNCTION__, array($item));
+        $call->arguments = array('<CacheItem>', $key, $value);
+        $this->addCall($call);
+
+        return $call->result;
+    }
+
+    public function getItems(array $keys = array())
+    {
+        $call = $this->timeCall(__FUNCTION__, array($keys));
+        $result = $call->result;
+        $call->result = sprintf('<DATA:%s>', gettype($result));
+        $this->addCall($call);
+
+        return $result;
+    }
+
+    public function clear()
+    {
+        $call = $this->timeCall(__FUNCTION__, array());
+        $this->addCall($call);
+
+        return $call->result;
+    }
+
+    public function deleteItems(array $keys)
+    {
+        $call = $this->timeCall(__FUNCTION__, array($keys));
+        $this->addCall($call);
+
+        return $call->result;
+    }
+
+    public function commit()
+    {
+        $call = $this->timeCall(__FUNCTION__);
+        $this->addCall($call);
+
+        return $call->result;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCalls()
+    {
+        return $this->calls;
+    }
+
+    /**
+     * Record a call.
+     *
+     * @param $call
+     */
+    private function addCall($call)
+    {
+        $this->calls[] = $call;
+    }
+
+    /**
+     * Get a string to represent the value.
+     *
+     * @param mixed $value
+     *
+     * @return string
+     */
+    private function getValueRepresentation($value)
```

Comment:
the representation should be handled by the data collector, using the cloneVar method of the base DataCollector